ContainerBuild.md

```diff
@@ -0,0 +1,167 @@
+# Build Ceph within a Container
+
+The Ceph project includes a script and additional files that help build
+the ceph.git sources inside an OCI-style container. This script requires
+Python 3 (tested with 3.8 or later) and podman or docker.
+
+Generally, the script aims to make it simple for a developer or anyone
+wanting to compile Ceph on Linux but not wanting to dedicate a full
+physical or virtual host to the job. The containers encapsulate the
+build dependencies and environment and are quick and easy to clean
+up when you no longer need them.
+
+# build-with-containers.py Introduction
+
+The script exists within the ceph.git tree at
+`src/script/build-with-container.py` and assumes it is being executed from
+within that tree. At any time the `--help` option can be passed to
+the script for a complete listing of command line options.
+
+To start with the default options run `./src/script/build-with-conttainer.py`.
+The script will first check if a build image already exists, if not it
+will construct one using the `Dockerfile.build` file in the ceph tree [1].
+
+Once a build container is available the script will build ceph from source using
+a build directory named `build` in the root of the source tree. The default
+environment uses a base distribution of `centos9`.
+
+You can select what base distribution to use with the `--distro`/`-d` option.
+The list of available distro bases include `ubuntu22.04` and `centos9`. These
+choices are known to work well. Other more experimental choices are available
```

Comment:
What are the experimental options?

Reply:
ubuntu 24.04 and centos 10 - text updated.